### PR TITLE
Modularize da precond

### PR DESCRIPTION
**Pull request by xukai92**
Addressing #259

### CODE REVIEW

**Comment by xukai92** [comment]
Cario build failed on AppVeyor

**Comment by xukai92** [comment]
Pushed few new commits on addressing #206.

File: src/samplers/support/adapt.jl [hunk 1]
```diff
@@ -0,0 +1,60 @@
+adapt_step_size(spl::Sampler, stats::Float64, δ::Float64) = begin
```
Comment (yebai) [inline src/samplers/support/adapt.jl]: maybe use a more specific sampler type, e.g. `Sampler{Hamiltonian}`?

Comment (yebai) [inline src/samplers/support/adapt.jl]: also for other adaption functions

Comment (xukai92) [inline src/samplers/support/adapt.jl]: Sure, let me do this.

**Pull request merged**